File: 백준/Silver/10828. 스택/스택.py
import sys

stack = []

n = int(input())  # 명령의 수

for _ in range(n):
    command = sys.stdin.readline().split()

    if command[0] == "push":
        stack.append(int(command[1]))
    elif command[0] == "pop":
        if stack:
            print(stack.pop())
        else:
            print(-1)
    elif command[0] == "size":
        print(len(stack))
    elif command[0] == "empty":
        if stack:
            print(0)
        else:
            print(1)
    elif command[0] == "top":
        if stack:
            print(stack[-1])
        else:
            print(-1)

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(br.readLine().trim());
        Deque<Integer> stack = new ArrayDeque<>(); // Stack보다 빠름

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken(); // 첫 번째 토큰 → 명령어((push, pop, ...))

            if (cmd.equals("push")) {
                int x = Integer.parseInt(st.nextToken()); // ((두 번째 토큰 → 숫자 3 ))
                stack.push(x);
            } else if (cmd.equals("pop")) {
                sb.append(stack.isEmpty() ? -1 : stack.pop()).append('\n');
            } else if (cmd.equals("size")) {
                sb.append(stack.size()).append('\n');
            } else if (cmd.equals("empty")) {
                sb.append(stack.isEmpty() ? 1 : 0).append('\n');
            } else if (cmd.equals("top")) {
                sb.append(stack.isEmpty() ? -1 : stack.peek()).append('\n');
            }
        }

        System.out.print(sb.toString());
    }
}
